packages/components/src/mobile/keyboard-avoiding-view/index.ios.js

```diff
@@ -3,18 +3,74 @@
  */
 import {
 	KeyboardAvoidingView as IOSKeyboardAvoidingView,
+	Animated,
+	Keyboard,
 	Dimensions,
 } from 'react-native';
 
-export const KeyboardAvoidingView = ( { parentHeight, ...otherProps } ) => {
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, useRef } from '@wordpress/element';
+
+const AnimatedKeyboardAvoidingView = Animated.createAnimatedComponent(
+	IOSKeyboardAvoidingView
+);
+
+const MIN_HEIGHT = 44;
+
+export const KeyboardAvoidingView = ( {
+	parentHeight,
+	style,
+	withAnimatedHeight = false,
+	...otherProps
+} ) => {
+	const [ keyboardHeight, setKeyboardHeight ] = useState( 0 );
+	const animatedHeight = useRef( new Animated.Value( MIN_HEIGHT ) ).current;
+
 	const { height: fullHeight } = Dimensions.get( 'window' );
 	const keyboardVerticalOffset = fullHeight - parentHeight;
 
+	useEffect( () => {
+		Keyboard.addListener( 'keyboardWillShow', onKeyboardWillShow );
+		Keyboard.addListener( 'keyboardWillHide', onKeyboardWillHide );
+		return () => {
+			Keyboard.removeListener( 'keyboardWillShow', onKeyboardWillShow );
+			Keyboard.removeListener( 'keyboardWillHide', onKeyboardWillHide );
+		};
+	}, [] );
+
+	useEffect( () => {
+		animate();
+	}, [ keyboardHeight ] );
+
+	function onKeyboardWillShow( { endCoordinates } ) {
+		setKeyboardHeight( endCoordinates.height );
+	}
+
+	function onKeyboardWillHide() {
+		setKeyboardHeight( 0 );
+	}
+
+	const paddedKeyboardHeight =
+		keyboardHeight + MIN_HEIGHT - ( style.bottom || 0 );
+
+	function animate() {
+		Animated.timing( animatedHeight, {
+			toValue: keyboardHeight ? paddedKeyboardHeight : MIN_HEIGHT,
+			duration: keyboardHeight ? 0 : 150,
+		} ).start();
```

Comment:
Can we add `useNativeDriver: false` ? In the [next version](https://github.com/react-native-community/releases/blob/master/CHANGELOG.md#v0620) (0.62), it'll be required so this way we'll avoid a warning in the upcoming future 😄